<launch>
    <arg name="speed" default="0"/>
    <arg name="steering_speed" default="0"/>
    
    <arg name="kp" default="5" />
    <arg name="ki" default="1.2" />
    <arg name="kd" default="0.4" />

    <arg name="turn_kp" default="10" />
    <arg name="turn_ki" default="20" />
    <arg name="turn_kd" default="0.1" />
    
    <group ns="raceon">
        <param name="topic_name_camera_image" value="camera/image" />
        <param name="topic_name_camera_info" value="camera/camera_info" />
        <param name="topic_name_position_pose" value="position/pose" />
        <param name="topic_name_position_track" value="position/track" />
        <param name="topic_name_manual_mode" value="control/manual_mode" />
        <param name="topic_name_control" value="control" />
        <param name="topic_name_encoder" value="encoder" />

	<!--node pkg="rosbag" type="record" name="record" output="screen" args="/raceon/control /raceon/position/pose"/-->

        <node pkg="raceon" name="actuation" type="actuation.py" output="log" />

        <node pkg="raceon" name="encoder" type="encoder.py" output="log" />

        <node pkg="raceon" name="camera" type="camera.py" output="screen">
            <param name="~calibration_file" value="$(find raceon)/calibration/undistort_params.pkl" />
            <param name="~publish_raw" value="false" />
            <param name="~resolution/width" value="640" />
            <param name="~resolution/height" value="480" />
            <param name="~fps" value="30" />
        </node>

        <node pkg="raceon" name="pos_estimation" type="pos_estimation.py" output="log">
            <param name="~scan_line_d" value="120" />
            <param name="~scan_line_u" value="80" />
            <param name="~peak_threshold_d" value="170" />
            <param name="~peak_threshold_u" value="170" />
            <param name="~track_width" value="470" />
            <param name="~camera_center" value="320" />
            <param name="~breaker" value="0" />
        </node>

        <node pkg="raceon" name="control" type="control.py" output="log">
            <param name="~motor_speed" value="$(arg speed)" />
            <param name="~steering_speed" value="$(arg steering_speed)" />
            <param name="~target" value="0" />
            <param name="~kp" value="$(arg kp)" />
            <param name="~ki" value="$(arg ki)" />
            <param name="~kd" value="$(arg kd)" />
            <param name="~turn_kp" value="$(arg turn_kp)" />
            <param name="~turn_ki" value="$(arg turn_ki)" />
            <param name="~turn_kd" value="$(arg turn_kd)" />
        </node>
    </group>
</launch>
